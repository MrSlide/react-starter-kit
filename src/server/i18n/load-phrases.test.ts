import loadPhrases from './load-phrases'

jest.mock('../constants/paths', function () {
  return {
    TRANSLATION_ASSETS_PATH: '/translations'
  }
})

jest.mock('../utils/fs', function () {
  return {
    find: function () {
      return [
        'en/test-a.json',
        'en/test-b.json',
        'en-gb/test-a.json',
        'pt-pt/test-a.json',
        'pt-pt/test-b.json'
      ]
    },
    readFile: async function (asset) {
      switch (asset) {
        case '/translations/en/test-a.json':
          return JSON.stringify({
            greeting: 'Hello, World',
            test: 'This is a test'
          })
        case '/translations/en/test-b.json':
          return JSON.stringify({
            foo: 'Foo',
            bar: 'Bar'
          })
        case '/translations/en-gb/test-a.json':
          return JSON.stringify({
            greeting: 'Good day, World'
          })
        case '/translations/pt-pt/test-a.json':
          return JSON.stringify({
            greeting: 'Olá, Mundo',
            test: 'Isto é um teste'
          })
        case '/translations/pt-pt/test-b.json':
          return JSON.stringify({
            foo: 'Foo',
            bar: 'Bar'
          })
      }
    }
  }
})

describe('i18n phrase loading', function () {
  test('loads all phrases with extended translations by default', async function () {
    const output = await loadPhrases()

    expect(output).toMatchSnapshot()
  })

  test('loads all phrases with extended translations if given an empty list of enabled languages', async function () {
    const output = await loadPhrases([])

    expect(output).toMatchSnapshot()
  })

  test('loads the enabled language phrases with extended translations', async function () {
    const output = await loadPhrases(['en-GB'])

    expect(output).toMatchSnapshot()
  })

  test('freezes the phrases', async function () {
    const output = await loadPhrases()

    expect(Object.isFrozen(output)).toBe(true)
  })
})
