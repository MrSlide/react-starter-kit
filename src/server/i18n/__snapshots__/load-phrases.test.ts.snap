// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i18n phrase loading loads all phrases with extended translations by default 1`] = `
Object {
  "en": Object {
    "testA": Object {
      "greeting": "Hello, World",
      "test": "This is a test",
    },
    "testB": Object {
      "bar": "Bar",
      "foo": "Foo",
    },
  },
  "en-GB": Object {
    "testA": Object {
      "greeting": "Good day, World",
      "test": "This is a test",
    },
    "testB": Object {
      "bar": "Bar",
      "foo": "Foo",
    },
  },
  "pt-PT": Object {
    "testA": Object {
      "greeting": "Olá, Mundo",
      "test": "Isto é um teste",
    },
    "testB": Object {
      "bar": "Bar",
      "foo": "Foo",
    },
  },
}
`;

exports[`i18n phrase loading loads all phrases with extended translations if given an empty list of enabled languages 1`] = `
Object {
  "en": Object {
    "testA": Object {
      "greeting": "Hello, World",
      "test": "This is a test",
    },
    "testB": Object {
      "bar": "Bar",
      "foo": "Foo",
    },
  },
  "en-GB": Object {
    "testA": Object {
      "greeting": "Good day, World",
      "test": "This is a test",
    },
    "testB": Object {
      "bar": "Bar",
      "foo": "Foo",
    },
  },
  "pt-PT": Object {
    "testA": Object {
      "greeting": "Olá, Mundo",
      "test": "Isto é um teste",
    },
    "testB": Object {
      "bar": "Bar",
      "foo": "Foo",
    },
  },
}
`;

exports[`i18n phrase loading loads the enabled language phrases with extended translations 1`] = `
Object {
  "en-GB": Object {
    "testA": Object {
      "greeting": "Good day, World",
      "test": "This is a test",
    },
    "testB": Object {
      "bar": "Bar",
      "foo": "Foo",
    },
  },
}
`;
